sw/device/silicon_creator/rom/e2e/BUILD

```
@@ -1990,61 +1990,144 @@ test_suite(
+        ":empty_test_slot_a_corrupted": {
             "key": "test_key_0",
-            "offset": offset,
+            "offset": SLOTS["a"],
+        },
+        ":empty_test_slot_b_corrupted": {
+            "key": "test_key_0",
+            "offset": SLOTS["b"],
+        },
+    },
+    devices = ["fpga_cw310"],
+)
+
+# Since we cannot feed the `assemble_flash_image` rule that is instantiated by
+# the `opentitan_multislot_flash_binary` macro an empty dictionary, we create
+# two images with "nothing" in them by created files of all ones, and stitching
+# them together.
+[genrule(
+    name = "sigverify_always_img_{}_nothing_{}_bin_signed_{}".format(slot, device, key),
+    outs = ["sigverify_always_img_all_ones_{}_bin_signed_{}".format(slot, device, key)],
+    cmd_bash = "dd if=/dev/zero bs=1024 count=1 status=none | tr \'\\0\' \'\\377\' | dd of=$(OUTS) bs=1024 status=none",
+) for slot in SLOTS for device in ["fpga_cw310"] for key in ["test_key_0"]]
+
+opentitan_multislot_flash_binary(
+    name = "sigverify_always_img_a_nothing_b_nothing",
+    srcs = {
+        ":sigverify_always_img_a_nothing": {
+            "key": "test_key_0",
+            "offset": SLOTS["a"],
+        },
+        ":sigverify_always_img_b_nothing": {
+            "key": "test_key_0",
+            "offset": SLOTS["b"],
         },
     },
     devices = ["fpga_cw310"],
-) for slot, offset in SLOTS.items()]
+)
+
+[otp_image(
+    name = "otp_img_sigverify_always_{}".format(lc_state.lower()),
+    src = "//hw/ip/otp_ctrl/data:otp_json_{}".format(lc_state.lower()),
+    overlays = STD_OTP_OVERLAYS,
+) for lc_state in structs.to_dict(CONST.LCV)]
+
+# Splice OTP images into bitstreams
+[
+    bitstream_splice(
+        name = "bitstream_sigverify_always_{}".format(lc_state.lower()),
+        src = "//hw/bitstream:rom",
+        data = ":otp_img_sigverify_always_{}".format(lc_state.lower()),
+        meminfo = "//hw/bitstream:otp_mmi",
+        tags = ["vivado"],
+        update_usr_access = True,
+    )
+    for lc_state in structs.to_dict(CONST.LCV)
+]
 
 SIGVERIFY_BAD_CASES = [
     {
-        "desc": "bad",
-        "suffix": "_corrupted",
+        "a": "nothing",
+        "b": "bad",
+        "expected_bfv": hex(CONST.BFV.SIGVERIFY.BAD_ENCODED_MSG),
+    },
+    {
+        "a": "bad",
+        "b": "nothing",
+        "expected_bfv": hex(CONST.BFV.SIGVERIFY.BAD_ENCODED_MSG),
+    },
+    {
+        "a": "bad",
+        "b": "bad",
+        "expected_bfv": hex(CONST.BFV.SIGVERIFY.BAD_ENCODED_MSG),
     },
     {
-        "desc": "empty",
-        "suffix": "_empty",
+        "a": "nothing",
+        "b": "nothing",
+        "expected_bfv": hex(CONST.BFV.BOOT_POLICY.BAD_IDENTIFIER),
     },
 ]
 
 [
     opentitan_functest(
-        name = "sigverify_always_a_{}_b_{}".format(
-            a["desc"],
-            b["desc"],
+        name = "sigverify_always_a_{}_b_{}_{}".format(
+            case["a"],
+            case["b"],
+            lc_state.lower(),
```

```suggestion
        name = "sigverify_always_{}_a_{}_b_{}".format(
            lc_state.lower(),
            case["a"],
            case["b"],
```

(faster execution if the tests end up sorted by name)

---

oh neat! thanks. fixed.